pkg/scheduler/scheduler.go

```diff
@@ -267,19 +267,40 @@ func (s *Scheduler) getScheduleType(key string) ScheduleType {
 			return FirstSchedule
 		}
 
-		policyNamespace := util.GetLabelValue(resourceBinding.Labels, util.PropagationPolicyNamespaceLabel)
-		policyName := util.GetLabelValue(resourceBinding.Labels, util.PropagationPolicyNameLabel)
+		var placement policyv1alpha1.Placement
+		var clusterPolicyName string
+		var policyName string
+		var policyNamespace string
+		if clusterPolicyName = util.GetLabelValue(resourceBinding.Labels, util.ClusterPropagationPolicyLabel); clusterPolicyName != "" {
+			clusterPolicy, err := s.clusterPolicyLister.Get(clusterPolicyName)
+			if err != nil {
+				return Unknown
+			}
 
-		policy, err := s.policyLister.PropagationPolicies(policyNamespace).Get(policyName)
-		if err != nil {
-			return Unknown
+			placement = clusterPolicy.Spec.Placement
 		}
-		placement, err := json.Marshal(policy.Spec.Placement)
+
+		if policyName = util.GetLabelValue(resourceBinding.Labels, util.PropagationPolicyNameLabel); policyName != "" {
+			policyNamespace = util.GetLabelValue(resourceBinding.Labels, util.PropagationPolicyNamespaceLabel)
+			policy, err := s.policyLister.PropagationPolicies(policyNamespace).Get(policyName)
+			if err != nil {
+				return Unknown
+			}
+
+			placement = policy.Spec.Placement
+		}
+
+		placements, err := json.Marshal(placement)
 		if err != nil {
-			klog.Errorf("Failed to marshal placement of propagationPolicy %s/%s, error: %v", policy.Namespace, policy.Name, err)
+			if clusterPolicyName != "" {
+				klog.Errorf("Failed to marshal placement of clusterPropagationPolicy %s, error: %v", clusterPolicyName, err)
+			} else {
+				klog.Errorf("Failed to marshal placement of propagationPolicy %s/%s, error: %v", policyNamespace, policyName, err)
+			}
+
 			return Unknown
 		}
-		policyPlacementStr := string(placement)
+		policyPlacementStr := string(placements)
```

Comment:
Can we abstract a method to get the `placementStr`?

Reply:
Then we can have unit testing to cover and more readable.

Reply:
make sense, please take a look the new version @RainbowMango